cpp/arcticdb/storage/s3/detail-inl.hpp

```diff
@@ -240,8 +241,19 @@ namespace s3 {
 
             util::check(to_delete.empty(), "Have {} segment that have not been removed",
                         to_delete.size());
-            if (!failed_deletes.empty())
-                throw KeyNotFoundException(Composite<VariantKey>(std::move(failed_deletes)));
+            if (!failed_deletes.empty()){
+                auto failed_deletes_message = std::ostringstream();
+                for (auto i=0u; i<failed_deletes.size(); ++i){
+                    auto& failed = failed_deletes[i];
+                    failed_deletes_message<<fmt::format("'{}' failed with '{}'", to_serialized_key(failed.first), failed.second);
+                    if (i != failed_deletes.size()){
+                        failed_deletes_message<<", ";
+                    }
+                }
+                auto error_message = fmt::format("Failed to delete some of the objects: {}.",
+                                                 failed_deletes_message.str());
+                raise<ErrorCode::E_UNEXPECTED_S3_ERROR>(error_message);
```

Comment:
Some of the keys not being present might still be the root cause right? In which case `KeyNotFoundException` would still be suitable. I'm concerned how we will change the catch-es further up the stack from `std::exception` to `KeyNotFoundException` with this change. Is there any way to get more info from AWS SDK about the root cause of the failures?

Reply:
I mean eventually we'll want to make the change to `version_map.hpp` here https://github.com/man-group/ArcticDB/pull/584/files#diff-d6219b28b5b4e3cb754fc3fd7b77f5faddb5c965ca27191157cd16018baf939d

Reply:
Resolving - Ivo explained that AWS SDK specifically states that deleting a non-existent object does not raise